src/configurationProvider.ts

```diff
@@ -33,6 +21,27 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return this.heuristicallyResolveDebugConfiguration(folder, config);
     }
 
+    private async provideDebugConfigurationsAync(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken) {
+        const res = <any[]>(await resolveMainClass());
+        const launchConfigs = res.map((item) => {
+            return {
+                type: "java",
+                name: "Debug (Launch)" + "-" + item.mainClass + "-" + item.projectName,

```

Comment:
change name to `baseName of item.mainClass <item.projectName>`. because the dropdown box is very narrow, show a simple class name is enough.

Reply:
what about different projects have a class with same name?